src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -186,12 +186,17 @@ internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpres
             }
             // PROTOTYPE(NullableReferenceTypes): Could we track nullability always,
             // even in C#7, but only report warnings when the feature is enabled?
-            var isNullable = includeNullability && !type.IsErrorType() ? expr.IsNullable() : null;
+            var isNullable = includeNullability ? expr.IsNullable() : null;
             return TypeSymbolWithAnnotations.Create(type, isNullable);
         }
 
         internal static bool? IsNullable(this BoundExpression expr)
         {
+            var type = expr.Type;
+            if ((object)type == null || type.IsErrorType())
```

Comment:
>if ((object)type == null || type.IsErrorType()) [](start = 12, length = 47)

I think we have other mechanism how to prevent accessing type of a variable when it is used in its own initializer. I think we should use that mechanism and target situation explicitly rather than basing the logic on whether the type is an error type or not .

Reply:
Fixed.